bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/handler/MideaACHandler.java

```diff
@@ -0,0 +1,1483 @@
+     * The check is for the IP, port and deviceID. This method also resets the dropped
+     * commands, disconnects the socket and stops the connection monitor (if these were
+     * running)
+     */
+    @Override
+    public void initialize() {
+        connectionManager.disconnect();
+        getConnectionManager().cancelConnectionMonitorJob();
+        connectionManager.resetDroppedCommands();
+        connectionManager.updateChannel(DROPPED_COMMANDS, new DecimalType(connectionManager.getDroppedCommands()));
+
+        config = getConfigAs(MideaACConfiguration.class);
+
+        setCloudProvider(CloudProviderDTO.getCloudProvider(config.cloud));
+        setSecurity(new Security(cloudProvider));
+
+        logger.debug("MideaACHandler config for {} is {}", thing.getUID(), config);
+
+        if (!config.isValid()) {
+            logger.warn("Configuration invalid for {}", thing.getUID());
+            if (config.isDiscoveryNeeded()) {
+                logger.warn("Discovery needed, discovering....{}", thing.getUID());
+                updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_PENDING,
+                        "Configuration missing, discovery needed. Discovering...");
+                MideaACDiscoveryService discoveryService = new MideaACDiscoveryService();
+
+                try {
+                    discoveryService.discoverThing(config.ipAddress, this);
+                } catch (Exception e) {
+                    logger.error("Discovery failure for {}: {}", thing.getUID(), e.getMessage());
+                }
+                return;
+            } else {
+                logger.debug("MideaACHandler config of {} is invalid. Check configuration", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Invalid MideaAC config. Check configuration.");
+                return;
+            }
+        } else {
+            logger.debug("Configuration valid for {}", thing.getUID());
+        }
+
+        ipAddress = config.ipAddress;
+        ipPort = config.ipPort;
+        deviceId = config.deviceId;
+        version = Integer.parseInt(config.version);
+
+        logger.debug("IPAddress: {}", ipAddress);
+        logger.debug("IPPort: {}", ipPort);
+        logger.debug("ID: {}", deviceId);
+        logger.debug("Version: {}", version);
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        connectionManager.connect();
+    }
+
+    @Override
+    public void discovered(DiscoveryResult discoveryResult) {
+        logger.debug("Discovered {}", thing.getUID());
+        Map<String, Object> discoveryProps = discoveryResult.getProperties();
+        Configuration configuration = editConfiguration();
+
+        @Nullable
+        Object propertyDeviceId = discoveryProps.get(CONFIG_DEVICEID);
+        if (propertyDeviceId != null) {
+            configuration.put(CONFIG_DEVICEID, propertyDeviceId.toString());
+        }
```

Comment:
Less code, more readable? 
```suggestion
        Object propertyDeviceId = Objects.requireNonNull(discoveryProps.get(CONFIG_DEVICEID));
```

Reply:
Done